src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -416,10 +423,51 @@ void Terminal::UpdateAppearance(const ICoreAppearance& appearance)
     // before, and shouldn't be now either.
     _scrollOffset = originalOffsetWasZero ? 0 : static_cast<int>(::base::ClampSub(_mutableViewport.Top(), newVisibleTop));
 
+    // Now that we've finished the hard work of resizing the main buffer and
+    // getting the viewport back into the right spot, we need to ALSO resize the
+    // alt buffer, if one exists. Fortunately, this is easy. We don't need to
+    // worry about the viewport and scrollback at all! The alt buffer never has
+    // any scrollback, so we just need to resize it and presto, we're done.

```

Comment:
It concerns me that this is not the same way that conhost is implemented. As I understand it, when the alt buffer is active, conhost doesn't resize the main buffer - that only happens when returning to the main buffer. Doing it this way, you could potentially be losing data in the main buffer that wouldn't otherwise have happened.

And in a perfect world, this resizing code would one day be shared between conhost and WT. The structures we're manipulating here are essentially the same in both code bases. I don't expect them to be merged at this point, but it would be preferable if they at least followed the same basic algorithm.

Reply:
That's a good point. Shouldn't be too hard to add in a `std::optional<til::size> _deferredResize` in `Terminal` and not process that until we're going back to the main buffer.

Reply:
wait `SCREEN_INFORMATION::SetViewportSize`. Hm. Looks like in the alt buffer we still resize the main buffer immediately. I could have swore we didn't though...

Reply:
Ah. I see what it is. 

SCREEN_INFO stores _window-based_ resizes for later, when in the alt buffer. It stashes the _HWND RECT itself_, and then processes that when done resizing. 

When we're doing conpty resizes though, we don't hit that path. We actually just resize the main buffer immediately. We may want to defer those as well. (What fortuitous timing I was in the middle of debugging why my fix for #3493 flickers in the Terminal)

Reply:
Hmmm... I never noticed that. It looks like there are a few things that could trigger a resize of some sort, and some resize the main buffer immediately and some don't. And just to confuse things even more, note that viewport resizing doesn't necessarily generate a buffer resize in conhost, because they can be independent. But hopefully WT doesn't need to worry about that yet.

Reply:
I might just end up rolling my #3493 branch into this one as well. Not sure I can detangle the two....


Actually, after typing it out - I'm gonna leave the resize as is in this PR. It's wacky, but I'm just gonna immediately replace it. I would however prefer to keep the PRs slightly more atomic.